include/envoy/singleton/manager.h

```diff
@@ -45,7 +45,7 @@ template <const char* name_param> class RegistrationImpl : public Registration {
  */
 #define SINGLETON_MANAGER_REGISTRATION(NAME)                                                       \
   static constexpr char NAME##_singleton_name[] = #NAME "_singleton";                              \
-  static Envoy::Registry::RegisterFactory<                                                         \
+  static Envoy::Registry::RegisterFactory< /* NOLINT(fuchsia-statically-constructed-objects) */    \
```

Comment:
@mattklein123 we can do sth like this for extensions if we use macro to register, WDYT? cc @htuch

Reply:
I guess? In general this strikes me as solving a problem that doesn't actually need solving, but I guess if we insist on not having any non-POD statics in the code base we have to do ithis?